src/elements/Asset.php

```diff
@@ -1888,16 +1888,7 @@ public function getUrlsBySize(array $sizes, mixed $transform = null): array
 
             [$value, $unit] = Assets::parseSrcsetSize($size);
 
-            $sizeTransform = $transform ? $transform->toArray([
-                'format',
-                'height',
-                'interlace',
-                'mode',
-                'position',
-                'quality',
-                'width',
-                'fill',
-            ]) : [];
```

Comment:
Getting all attributes instead of specifying, so we pick up any defined by a different class.

Alternatively, we could introduce something like `\craft\models\ImageTransform::getImageAttributes` that returned the attribute names involved in the actual image manipulation.